lib/Client/MultiCurl.php

```diff
@@ -90,8 +116,31 @@ public function proceed(): void
             return;
         }
 
-        if (!$this->curlm && false === $this->curlm = curl_multi_init()) {
-            throw new ClientException('Unable to create a new cURL multi handle');
+        if (!$this->curlm) {
+            if (false === $this->curlm = curl_multi_init()) {
+                throw new ClientException('Unable to create a new cURL multi handle');
+            }
+
+            $userCallbacks = $this->pushFunctions;
+            $cb = function ($parent, $pushed, $headers) use ($userCallbacks) {
+                // If any callback say no, then do not accept.
+                foreach ($userCallbacks as $callback) {
+                    if (CURL_PUSH_DENY === $callback($parent, $pushed, $headers)) {
+                        return CURL_PUSH_DENY;
+                    }
+                }
+
+                curl_setopt($pushed, CURLOPT_RETURNTRANSFER, true);
+                curl_setopt($pushed, CURLOPT_HEADER, true);
+                curl_setopt($pushed, CURLOPT_HEADERFUNCTION, null);
+                curl_setopt($pushed, CURLOPT_WRITEFUNCTION, null);
```

Comment:
These options do not seam to have much effect.
Doing curl_setopt($pushed, CURLOPT_WRITEFUNCTION, null); does not help...